lib/spack/spack/build_environment.py

```diff
@@ -1339,26 +1370,44 @@ def child_fun():
         if input_fd is not None:
             input_fd.close()
 
-    def exitcode_msg(p):
-        typ = "exit" if p.exitcode >= 0 else "signal"
-        return f"{typ} {abs(p.exitcode)}"
+    # Create an instance of ProcessHandle with the child processes'
```

Comment:
```suggestion
    # Create an instance of ProcessHandle with the child process's
```

Could also just make this comment a little higher level, e.g., "Create a ProcessHandle that the caller can use to track and complete the process started by this function".  The parameters, etc. can change but the general idea likely won't for a while.